5584 Updated program icons/styles

CarlosNZ opened the pull request:
<!-- IMPORTANT!
  - Every PR must reference an issue; this helps to explain the intent of the PR
 -->

Fixes #5584

Replaces the existing icons in `medical-icons.css` with new ones supplied by Maria, and changed the styling to match the mockups. Have only eyeballed it though, haven't referred to any particular specs, assume this is okay?

<img width="1034" alt="Screenshot 2024-12-02 at 5 02 23 PM" src="https://github.com/user-attachments/assets/bab0387a-7276-48d3-b807-28a849dd1771">
<img width="1081" alt="Screenshot 2024-12-02 at 5 02 28 PM" src="https://github.com/user-attachments/assets/532e1b7f-3b8a-4ed6-8847-b2dbf7b8a509">


## 💌 Any notes for the reviewer?

I've made these new icons standard React SVG icons, rather than the "css" approach of the old ones. Seems easier to manage and maintain. I presume the reason it was done like that before is so icons could be specified dynamically from program schemas, but I've done this a different way now, which seems a bit more consistent with our normal patterns.

I've also specified a fallback icon -- the "stethescope" one, which seems like a suitable all-purpose medical icon when not specifed.

I think there will be some updates required for the existing programs schemas, but I've tried to minimise this.

# 🧪 Testing

<!-- Explain the steps you'd take to test the changes of this PR manually -->

- [ ] Have some programs configured
- [ ] Go to, or create and encounter
- [ ] Notice the new icon buttons for Encounters (and Contact tracing)

jmbrunskill commented:
> I think there will be some updates required for the existing programs schemas, but I've tried to minimise this.

Can you clarify what/why? Do the names change?
I'm wondering if we should keep the old css around for legacy support?

jmbrunskill reviewed (commented):
I like the new approach, but would like to have a clear plan about what might break with existing forms.

@mark-prins and comments on this approach?

mark-prins commented on client/packages/common/src/ui/medical-icons/Contraceptive.tsx:
```diff
@@ -0,0 +1,26 @@
+        y2="33.7131"
+        gradientUnits="userSpaceOnUse"
+      >
+        <stop stop-color="#E73735" />
```
If moving to components, please use the theme rather than hard coded colours.
These are also colours outside of our current theme which isn't ideal. Perhaps should be primary.main and primary.light?

Just use `useTheme` to pull in the theme and take the colours as needed.

mark-prins commented:
The reason for using a css previously was to reduce the bundle size. Not everyone is using these icons and the css was 128k - putting into a static css allowed for local caching and meant that only users of encounters had to download it.

Also, what's happened to some of the icons?

![image](https://github.com/user-attachments/assets/b061a034-286a-4989-b6c4-ea44c800f35b)

we did have icons for all the areas.. but there's a lot of stethoscopes!

CarlosNZ commented:
> Can you clarify what/why? Do the names change?

I just checked and the only previous icon key that doesn't exist in the new ones is "injection". So I added another alias for it in the mapping object to map it to the ImmunizationIcon 103ed6b7fb

mark-prins commented on client/packages/common/src/ui/medical-icons/index.ts:
```diff
@@ -0,0 +1,83 @@
+  virus: VirusIcon,
+  viralLoad: VirusIcon,
+  visitType: VisitTypeIcon,
+  visit: VisitTypeIcon,
```
they aren't the same though!
`indexTesting` => `'index-testing'`
`documents` => `document`
`injection` is missing
`lungsVirus` => `'lungs-virus'`
`medicalBag` => `'medical-bag'`
`medicalClipboard` => `'medical-chart'`
`testTube` => `'test-tube'`
`gbv` => `'gender-based-violence'`

CarlosNZ replied:
Ah, my rushed regex search through that massive file missed the hyphenated ones 🙄

mark-prins commented:
can you also update the hover state please?
Just a background of `#D2DFFF` ( or whatever is closest in our theme.ts! )

![telegram-cloud-photo-size-1-5073822897910427109-x](https://github.com/user-attachments/assets/9ab045fa-bfef-4b45-9f13-1744e8fb6cf6)


So.. to summarise.. my requests 🙏 
* update the icon names
* update hover state
* use theme rather than hard coded colours

thanks! the icons look great btw, much nicer. and yes, good idea to use actual components, we can manage them more easily. just keep an eye on the bundle size plz

CarlosNZ commented:
> The reason for using a css previously was to reduce the bundle size. Not everyone is using these icons and the css was 128k - putting into a static css allowed for local caching and meant that only users of encounters had to download it.

Can we just put these icons in the "programs" pacakge?

mark-prins commented:
> Can you clarify what/why? Do the names change?

it's just that the JSON schema refers to the icons; if we rename the icons we need to find and update all schemas ( yes, it's not ideal.. )

> Can we just put these icons in the "programs" pacakge?

they're probably only useful in the programs package tbf. I'm pretty relaxed tho - these are only 50k or so? not too bad?

CarlosNZ commented:
Sorry @mark-prins, I've updated this with a bit more thoroughness:
- Moved icons to Programs package (and renamed them "ProgramIcons") (the total bundle is actually now 50K lighter 😉)
- Put the Gradient stops in a common component with theme.palette colours (common component saves having to do `useTheme` in every icon, plus makes it easier to change the gradient as well as the colours whenever we want)
- Fixed up the names so all existing ones get correctly mapped to a new icon (I've made camelCase and hyphen-case aliases for the multi-word names)
- updated hover style

mark-prins reviewed (approved):
👍  very nice! like the use of the <GradientStops /> 🙏

CarlosNZ commented:
Thanks @mark-prins!
